fv3gfs/physics/testing/translate_physics.py

```diff
@@ -0,0 +1,69 @@
+from fv3core.testing import TranslateFortranData2Py
+import numpy as np
+import fv3core.utils.gt4py_utils as utils
+
+
+def read_serialized_data(serializer, savepoint, variable):
+    data = serializer.read(variable, savepoint)
+    n_dim = len(data.shape)
+    cn = int(np.sqrt(data.shape[0]))
+    npz = data.shape[1]
+    if n_dim == 2:
+        var_reshape = np.reshape(data, (cn, cn, npz))
+        rearranged = var_reshape[:, :, ::-1]
+    elif n_dim == 3:
+        n_trac = data.shape[2]
+        var_reshape = np.reshape(data, (cn, cn, npz, n_trac))
+        rearranged = var_reshape[:, :, ::-1, :]
+    elif len(data.flatten()) == 1:
+        rearranged = data[0]
+    else:
+        raise NotImplementedError("Data dimension not supported")
+    return rearranged
```

Comment:
it works to do this, though since read_serialized_data doesn't belong to the class, but is meant to be super generic, one alternative option would be to incorporate this logic into collect_input_data. Then this logic belongs to TranslatePhysicsFortranData2Py instead floating on its own. especially since you also need to do the parallel of overriding slice_output, I think it makes sense there.

Reply:
Good point, moved the logic to collect_input_data in TranslatePhysicsFortranData2Py